charts/osm/templates/prometheus-configmap.yaml

```diff
@@ -111,6 +111,104 @@ data:
+        metric_relabel_configs:
+        - source_labels: [__name__]
+          regex: 'envoy_.*osm_request_(total|duration_ms_(bucket|count|sum))'
+          action: keep
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_(\d{3})_source_namespace_.*_source_kind_.*_source_name_.*_source_pod_.*_destination_namespace_.*_destination_kind_.*_destination_name_.*_destination_pod_.*_osm_request_total
+          target_label: response_code
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_(.*)_source_kind_.*_source_name_.*_source_pod_.*_destination_namespace_.*_destination_kind_.*_destination_name_.*_destination_pod_.*_osm_request_total
+          target_label: source_namespace
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_.*_source_kind_(.*)_source_name_.*_source_pod_.*_destination_namespace_.*_destination_kind_.*_destination_name_.*_destination_pod_.*_osm_request_total
+          target_label: source_kind
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_.*_source_kind_.*_source_name_(.*)_source_pod_.*_destination_namespace_.*_destination_kind_.*_destination_name_.*_destination_pod_.*_osm_request_total
+          target_label: source_name
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_.*_source_kind_.*_source_name_.*_source_pod_(.*)_destination_namespace_.*_destination_kind_.*_destination_name_.*_destination_pod_.*_osm_request_total
+          target_label: source_pod
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_.*_source_kind_.*_source_name_.*_source_pod_.*_destination_namespace_(.*)_destination_kind_.*_destination_name_.*_destination_pod_.*_osm_request_total
+          target_label: destination_namespace
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_.*_source_kind_.*_source_name_.*_source_pod_.*_destination_namespace_.*_destination_kind_(.*)_destination_name_.*_destination_pod_.*_osm_request_total
+          target_label: destination_kind
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_.*_source_kind_.*_source_name_.*_source_pod_.*_destination_namespace_.*_destination_kind_.*_destination_name_(.*)_destination_pod_.*_osm_request_total
+          target_label: destination_name
+        - source_labels: [__name__]
+          action: replace
+          regex: envoy_response_code_\d{3}_source_namespace_.*_source_kind_.*_source_name_.*_source_pod_.*_destination_namespace_.*_destination_kind_.*_destination_name_.*_destination_pod_(.*)_osm_request_total
```

Comment:
you are still probably making it too hard on yourself?

if you are really only interested in one label, probably the unique tokens/word before and after it are enough, everything else on the string can just be obviated/`.*`. I this right?
ie. 
```
envoy_response_code_\d{3}_source_namespace_.*_source_kind_.*_source_name_.*_source_pod_.*_destination_namespace_.*_destination_kind_.*_destination_name_.*_destination_pod_(.*)_osm_request_total
```
vs
```
.*_destination_pod_(.*)_osm_request_total.*
```

Reply:
Talked with @eduser25 offline

My main concern is making sure that the regexes handle weird cases like a pod named "source_namespace" and this was the easiest way to make sure those are handled properly.

I'll plan on keeping these as-is for now and then taking another pass at these later when I get a perf environment set up and can do more edge case testing.